drivers/dma/dma_intel_adsp_gpdma.c

```diff
@@ -190,16 +190,24 @@ static void intel_adsp_gpdma_clock_enable(const struct device *dev)
 	sys_write32(val, reg);
 }
 
-#ifdef CONFIG_SOC_SERIES_INTEL_ACE
 static void intel_adsp_gpdma_select_owner(const struct device *dev)
```

Comment:
Maybe if for feature and then if for implementation

if CONFIG_DMA_INTEL_ADSP_GPDMA_NEED_CONTROLLER_OWNERSHIP
--if SOC_A
--else or if SOC_B